### PR TITLE
feat(Empty): description prop support false value to remove description node

### PR DESCRIPTION
**Pull request by billfeller**
### 🤔 This is a ...

- [ ] Bug fix
- [x] Site / document update
- [x] New feature
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

in **Empty** Component we do not need description always. i can use **description={' '}** to do it, but i think  **description={false}** may be better. 

### 💡 Solution

In **Empty** Component, description prop support false value, just as: 
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Empty component support no description  |
| 🇨🇳 Chinese | 空状态支持无描述展示 |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/empty/demo/description.md](https://github.com/billfeller/ant-design/blob/master/components/empty/demo/description.md)

### CODE REVIEW

File: components/empty/index.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ const OriginEmpty: React.SFC<EmptyProps> = (props: EmptyProps) => (
         <LocaleReceiver componentName="Empty">
           {(locale: TransferLocale) => {
             const prefixCls = getPrefixCls('empty', customizePrefixCls);
-            const des = description || locale.description;
+            const des = typeof description === 'boolean' ? description : (description || locale.description);
```
Comment (afc163) [inline components/empty/index.tsx]:
```suggestion
            const des = typeof description !== 'undefined' ? description : locale.description;
```

Then the extra boolean type is no need to add.

Comment (ztplz) [reply components/empty/index.tsx]: 问题是 如果是false 它应该显示默认的  还是什么都不显示。。。
image
这是react默认的behavior

Comment (billfeller) [reply components/empty/index.tsx]: i remove the boolean type, and add checked whether description is nil to do it.

File: components/empty/index.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export interface EmptyProps {
    */
   imageStyle?: React.CSSProperties;
   image?: React.ReactNode;
-  description?: React.ReactNode;
+  description?: React.ReactNode | boolean;
```
[the remark this answers is not in the document]
Comment (billfeller) [reply components/empty/index.tsx]: i remove the boolean type, and add checked whether description is nil to do it.

File: components/empty/index.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ const OriginEmpty: React.SFC<EmptyProps> = (props: EmptyProps) => (
         <LocaleReceiver componentName="Empty">
           {(locale: TransferLocale) => {
             const prefixCls = getPrefixCls('empty', customizePrefixCls);
-            const des = description || locale.description;
+            const des = description == null ? description : (description || locale.description);
```
Comment (afc163) [inline components/empty/index.tsx]: `typeof description !== 'undefined'`

Comment (afc163) [reply components/empty/index.tsx]: 你误解我意思了。不写 description 就相当于 `description={undefined}`，这种情况走默认 locale.description。

`description={false}` `description={0}` `description=""` `description={null}` 这些空值，都不显示 `description`。

Comment (afc163) [reply components/empty/index.tsx]: 这样实现最简洁，涵盖所有类型。

Comment (billfeller) [reply components/empty/index.tsx]: 明白了，我再改下

Comment (billfeller) [reply components/empty/index.tsx]: 已修复，description主动赋空值，都不显示 `description`

Comment (billfeller) [reply components/empty/index.tsx]: `const des = typeof description !== 'undefined' ? description : locale.description;`

**Comment by afc163** [comment]
ci failed

File: components/empty/demo/description.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+order: 0
```
Comment (afc163) [inline components/empty/demo/description.md]: 不要放第一个，放到后面去。

Comment (billfeller) [reply components/empty/demo/description.md]: 构建问题，排序问题，已修复

**Pull request merged**